src/machine.rs

```diff
@@ -339,6 +339,9 @@ pub struct Evaluator<'mir, 'tcx> {
     pub(crate) report_progress: Option<u32>,
     /// The number of blocks that passed since the last progress report.
     pub(crate) since_progress_report: u32,
+
+    /// Path to the optional C shared object file
+    pub external_c_so_file: Option<String>,
```

Comment:
Maybe `Option<libloading::Library>` to avoid reloads?

Reply:
good idea!